crates/subspace-service/src/sync_from_dsn.rs

```diff
@@ -307,10 +307,10 @@ where
     // right away.
     // TODO: implement starting from any segment index, and handle the partial block at the end of
     // the last segment.
-    let mut last_processed_segment_index = SegmentIndex::ZERO;
+    let last_processed_segment_index = SegmentIndex::ZERO;
```

Comment:
Are they only not used if DSN sync is unable to restart?
And if it is needed, then probably the argument should be removed from `import_blocks_from_dsn` completely.